src/utility/HCISharedMemTransport.cpp

```diff
@@ -0,0 +1,837 @@
+#endif /* DEBUG */
+            break;
+    }
+
+    /*  In case Event belongs to the Evt Pool we need to inform  */
+    if (((uint8_t *)hcievt >= EvtPool) && ((uint8_t *)hcievt < (EvtPool + POOL_SIZE))) {
+        /*  Free the message from shared memory */
+        TL_MM_EvtDone(hcievt);
+    }
+}
+
+/**
+ * TL Mailbox synchronisation means
+ */
+
+/*  returns true if sys_event was received, false otherwise */
+static bool sysevt_wait(void)
+{
+    /* sys_event remains false until event is received */
+//    for (unsigned long start = millis(); (millis() - start) < 10000;) {
+    while(1) {
+        /* Wait for 10sec max - if not return an error */
+        if (sys_event) { break; }
+    }
+
+    if (!sys_event) {
+        /*  no event received, timeout occurs */
+         return false;
+    } else {
+        /*  release immmediately, now that M0 runs */
+        sys_event = false;
+        return true;
+    }
+}
+
+/*  WEAK callbacks from the BLE TL driver - will be called under Interrupt */
+static void sysevt_received(void *pdata)
+{
+    /* For now only READY event is received, so we know this is it */
+    sys_event = true;
+    /* But later on ... we'll have to parse the answer */
+    
+    return;
+}
+
+/*  returns true if sysevt was already received, which means M0 core is
+ *  already up and running */
+static bool sysevt_check(void)
+{
+    /*  Check if system is UP and runing already */
+//    for (unsigned long start = millis(); (millis() - start) < 10;) {
+    while(1) {
+        if (sys_event) { break; }
+    }
+
+    if (!sys_event) {
+        return false;
+    } else {
+        /*  release immmediately as M0 already runs */
+        sys_event = false;
+        return true;
+    }
+}
+
+void shci_cmd_resp_release(uint32_t flag)
+{
+    sys_resp = false;
+    return;
+}
+
+void shci_cmd_resp_wait(uint32_t timeout)
+{
+    /* TO DO: manage timeouts if we can return an error */
+//    for (unsigned long start = millis(); (millis() - start) < 10000;) {
+    while(1) {
+        /* Wait for 10sec max - if not return an error */
+        if (sys_resp) { break; }
+    }
+
+    if (!sys_resp) {
+        /*  no event received, timeout occurs */
+        return;
+    }
+}
+
+static void acl_data_ack(void)
+{
+    /**
+     * The current implementation assumes the tackGUI will not send a new HCI ACL DATA packet before this ack is received
+     * ( which means the CPU2 has handled the previous packet )
+     * In order to implement a secure mechanism, it is required either
+     * - a flow control with the stack
+     * - a local pool of buffer to store packets received from the stack
+     */
+    acl_data = false;
+    return;
+}
+
+static bool acl_data_wait(void)
+{
+    /* Wait 10 sec for previous ACL command to be ack'ed by Low Layers
+     * before sending the next one */
+//    for (unsigned long start = millis(); (millis() - start) < 10000;) {
+    while(1) {
```

Comment:
same here as well.
the goal is that only 1 ACL command is sent at a time until we receive the response. 
- 1st ACT DATA send request => send it 
- 2nd ACL DATA send request => wait for ACK
- ACK of 1st ACL DATA received
- OK send the 2nd ACL DATA REQ
- ACK of 2nd ACL DATA received
- etc

Reply:
this can now restore the timeout instead of infinite loop